server/schedule/filter/filters.go

```diff
@@ -295,66 +295,126 @@ func (f *StoreStateFilter) Type() string {
 type conditionFunc func(*config.PersistOptions, *core.StoreInfo) bool
 
 func (f *StoreStateFilter) isRemoved(opt *config.PersistOptions, store *core.StoreInfo) bool {
-	f.Reason = "tombstone"
-	return store.IsRemoved()
+	isRemoved := store.IsRemoved()
+	if isRemoved {
+		f.Reason = "tombstone"
+	} else {
```

Comment:
We can remove else?

Reply:
I don't think else can be removed. In function anyConditionMatch, any condition matches , it will return and keep the reason in Reason, but if none matches, anyConditionMatch return false, but Reason will keep the last reason before calling anyConditionMatch.